File: 121-Best-Time-to-Buy-and-Sell-Stock.js
const maxProfit = function(prices) {
    let max_till_now = prices[0]
    let min_till_now = prices[0]
    result = 0
    for (i in prices){
        if (prices[i] > max_till_now){
            max_till_now = prices[i] 
            result = max_till_now - min_till_now
        }
        if(prices[i] < min_till_now){
            min_till_now = prices[i]
            result = max_till_now - min_till_now
        }

    }
    console.log(max_till_now, min_till_now, result)
};

maxProfit([7,6,4,3,1])

File: 121-Best-Time-to-Buy-and-Sell-Stock.py
def maxProfit(prices):
    minSoFar = prices[0]
    maxProfit = 0
    for p in prices:
        minSoFar = min(p, minSoFar)
        profit = p - minSoFar
        maxProfit = max(profit, maxProfit)

    return maxProfit
        

print(maxProfit([2,4,3,5,1]))